### ISSUE TITLE
Feature request: Direct Connect support for changing network MTU

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#6136]
Support for changing a Direct Connect private virtual interface's MTU (AKA jumbo frames) [released](https://docs.aws.amazon.com/directconnect/latest/UserGuide/set-jumbo-frames-vif.html).
Add support to the various Direct Connect private virtual interface resources:
* [`aws_dx_private_virtual_interface`](https://www.terraform.io/docs/providers/aws/r/dx_private_virtual_interface.html)
* [`aws_dx_hosted_private_virtual_interface`](aws_dx_hosted_private_virtual_interface)

**Comment by ewbankkit** [comment]
Suggested syntax:

```hcl
resource "aws_dx_private_virtual_interface" "foo" {
  connection_id = "dxcon-zzzzzzzz"

  name           = "vif-foo"
  vlan           = 4094
  address_family = "ipv4"
  bgp_asn        = 65352
  mtu            = 9001 // Jumbo frames
}
```

**Comment by ewbankkit** [comment]
A new, computed  `jumbo_frame_capable` attribute will be added to the [`aws_dx_connection`](https://www.terraform.io/docs/providers/aws/r/dx_connection.html) and  private virtual interface resources.

**Comment by slapula** [comment]
@ewbankkit I hope you don't mind but I've went ahead and submitted PRs to implement this request.  Let me know if I missed anything.

**Comment by ewbankkit** [comment]
No problem at all, many thanks.
I left a few comments in the PRs.
Have you managed to run the acceptance tests? If so, please attach the outputs. Otherwise I can help out with that.

**Comment by slapula** [comment]
@ewbankkit I believe I've address your comments across all of my related PRs.  I have not been able to really run the acceptance tests due to the nature of the service (mainly I don't have a valid connection ID to test with in my personal account).

**Comment by ewbankkit** [comment]
@slapula Great stuff! I've added a couple more comments.
When you're ready I can run acceptance tests.

**Comment by ewbankkit** [comment]
@bflad Shall I reopen this one until #6141 and #6142 are merged or open 2 more issues?

**Comment by bflad** [comment]
Support for both of these resources has been merged and will release with version 1.42.0 of the AWS provider, likely today or tomorrow. 👍

**Comment by bflad** [comment]
This has been released in [version 1.42.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1420-october-31-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
